index.bs

```diff
@@ -516,6 +522,31 @@ controls the maximum number of attributions for a
 
 # General Algorithms # {#general-algorithms}
 
+<h3 id="parsing-filter-data">Parsing filter data</h3>
+
+To <dfn>parse filter data</dfn> given a |value| and an optional [=boolean=]
+<dfn for="parse filter data"><var>forSource</var></dfn> (default false):
+
+1. If |value| is not a [=map=], return null.
+1. If |forSource| is true and |value|["`source_type`"] [=map/exists=], return
+    null.
+1. If |value|'s [=map/size=] is greater than the user agent's
+    [=max entries per attribution filter map=], return null.
+1. Let |result| be a new [=attribution filter map=].
+1. [=map/iterate|For each=] |filter| → |data| of |value|:
+    1. If |data| is not a [=list=], return null.
+    1. Let |set| be a new [=ordered set=].
+    1. [=list/iterate|For each=] |d| of |data|:
```

Comment:
Do we want to use a more specific name here like filterValueItem? I'm fine with the short form d as well though.

Reply:
It has a short scope, so I think it's fine as is.